utils/connection_handler.py

```
@@ -0,0 +1,107 @@
+        """
+        Get connection statistics.
+        
+        Returns:
+            Dict with connection metrics
+        """
+        success_rate = ((self.total_calls - self.failed_attempts) / self.total_calls * 100) if self.total_calls > 0 else 100
+        
+        return {
+            'failed_attempts': self.failed_attempts,
+            'successful_retries': self.successful_retries,
+            'total_attempts': total_attempts,
```

Ruff (0.14.10)

Undefined name `total_attempts`

(F821)